app/src/main/java/com/google/jetpackcamera/MainActivity.kt

```diff
@@ -160,6 +162,11 @@ class MainActivity : ComponentActivity() {
         }
     }
 
+    private fun isDebugMode(): Boolean {
+        return intent != null && intent.hasExtra(KEY_DEBUG_MODE) &&
+            intent.getBooleanExtra(KEY_DEBUG_MODE, false)
+    }
+
```

Comment:
nit: No need to check if it has the extra if you're already providing a default. You can just do:
```
private fun isDebugMode(): Boolean =
    intent?.getBooleanExtra(KEY_DEBUG_MODE, false) ?: false
```

or as a property:
```
private val isDebugMode: Boolean
    get() = intent?.getBooleanExtra(KEY_DEBUG_MODE, false) ?: false
```